harmonize GDP and population

klau506 opened the pull request:
Add harmonized files and harmonization data, as well as the code used to produce them

klau506 committed:
harmonize GDP and population

Add harmonized files and code used to produce them

russellhz commented on input/gcamdata/inst/extdata/socioeconomics/GCAM3_GDP.csv:
```diff
@@ -4,17 +4,17 @@
-Southeast Asia,222332,593705,1230630,1460320,1961080,2601350,3372790,4305090,5403760,6689470,8210530,9959920,12061300,14448000,17201800,20092200,23375600,26794100,30544100,34509200,38874800,43534900
-USA,3473910,5728530,9104140,9600970,10756900,11914400,13175100,14458700,15829500,17465700,19164900,21251000,23198700,25517400,27872600,30669200,33583600,36744000,39970500,43838800,48155500,52760200
-Western Europe,3802400,5646240,7801260,7930550,8546440,9184410,9843660,10470000,11052000,11624200,12189200,12795600,13407800,14089100,14736400,15486200,16180100,16919800,17638600,18551700,19738200,20254600
+Africa,250238,372580,2285502,2935806,3471964,3806228,4649023,5868433,7687257,10008768,12969888,16749961,21576781,27668493,35171412,44298418,55210792,68055680,82916939,99846450,118969449,140339237
```
some of these changes (like the Africa GDP in 2005) are really big from the old file...any idea why? mostly concerned this may be a units issue

klau506 replied:
Yes, I saw it, but since other regions have similar values to the old file I don't think it's a matter of units... But I'll recheck again just in case

russellhz commented:
it looks like this will change `socioeconomics_GCAM3.xml`. Do you know how it will affect `socioeconomics_macro.xml`? I'm not sure what assumptions go into the new macro file and whether we want it to be aligned with this harmonization

klau506 committed:
Fix GDP harmonization

-Consider MER - PPP units change
-Change chunks to use GCAM3_GDP data
-Rename the final xml

russellhz commented:
sorry that I'm just now realizing this, but I think we need to use the SSP processing code for population and GDP. The SSP database uses country-level data, whereas GCAM3 uses GCAM3 regional data (but must do some downscaling to get to the 32 GCAM regions). Since the harmonization data is by country, it makes more sense for us to directly use the country data provided than for us to group to GCAM3-regions and then disaggregate to GCAM-32 region @klau506 @jonsampedro

klau506 committed:
TO BE FIXED: simplify socioeconomics' R chunks

GDP value not matching USDA or GCAM3 data. Population values work well

klau506 committed:
Use PPP-MER gcam conversion by iso

russellhz committed:
small changes to IC GDP

klau506 committed:
Use gdp_deflator by country

russellhz committed:
Revert "Use gdp_deflator by country"

This reverts commit 3796a4e9db76c34daf4007d578976fb87a381423.

jonsampedro commented:
No need for this pull request if we merge the other, right?
